locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/SharedLocationSecurityGroupCustomizer.java

```
@@ -0,0 +1,172 @@
+     * security groups
+     */
+    private int[] inboundPorts;
+
+    /**
+     * The location name is appended to the name of the shared SG - use if you need distinct shared SGs within the same location
+     *
+     * @param locationName appended to the name of the SG
+     */
+    public void setLocationName(String locationName) {
+        this.locationName = locationName;
+    }
+
+    /**
+     * Extra ports to be opened on the entities in the SG
+     *
+     * @param portRanges
+     */
+    public void setPortRanges(List<String> portRanges) {
+        this.portRanges = portRanges;
+    }
+
+    /**
+     * Set to restict the range that the ports that are to be opened can be accessed from
+     *
+     * @param cidr
+     */
+    public void setCidr(String cidr) {
+        this.cidr = cidr;
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, TemplateOptions templateOptions) {
+        super.customize(location, computeService, templateOptions);
+        inboundPorts = templateOptions.getInboundPorts();
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, Template template) {
+        super.customize(location, computeService, template);
+
+        final JcloudsLocationSecurityGroupCustomizer instance = getInstance(getSharedGroupId(location));
+
+        if (cidr != null) instance.setSshCidrSupplier(Suppliers.ofInstance(new Cidr(cidr)));
+
+        instance.customize(location, computeService, template);
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, JcloudsMachineLocation machine) {
+        super.customize(location, computeService, machine);
+
+        final JcloudsLocationSecurityGroupCustomizer instance = getInstance(getSharedGroupId(location));
+
+        if (portRanges != null) {
+            RangeSet<Integer> portRanges = BrooklynNetworkUtils.portRulesToRanges(this.portRanges);
```

I'd parse these when they are first set, rather than when they are about to be used - fail fast (rather than after someone has waited for their VM to be provisioned).